### PR TITLE
Support using cloud description controls

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**

### CODE REVIEW

File: resourcehandler/k8sresources.go [hunk 1]
```diff
@@ -68,6 +73,31 @@ func (k8sHandler *K8sResourceHandler) GetResources(frameworks []reporthandling.F
...
 }
 
+func getCloudProviderDescription(allResources map[string]workloadinterface.IMetadata, k8sResourcesMap *cautils.K8SResources) error {
+	if cloudsupport.IsRunningInCloudProvider() {
+		wl, err := cloudsupport.GetDescriptiveInfoFromCloudProvider()
+		if err != nil {
+			cluster := k8sinterface.GetCurrentContext().Cluster
+			provider := cloudsupport.GetCloudProvider(cluster)
+			// Return error with useful info on how to configure credentials for getting cloud provider info
+			switch provider {
```
Comment (dwertent) [inline resourcehandler/k8sresources.go:83]: Why is the package not handling the error logic?

File: resourcehandler/k8sresources.go [hunk 2]
```diff
@@ -68,6 +73,31 @@ func (k8sHandler *K8sResourceHandler) GetResources(frameworks []reporthandling.F
...
+			case "eks":
+				return fmt.Errorf("could not get descriptive information about eks cluster: %s using sdk client. Check out how to configure credentials in https://docs.aws.amazon.com/sdk-for-go/api/", cluster)
+			case "aks":
+				return fmt.Errorf("could not get descriptive information about aks cluster: %s. %v", cluster, err.Error())
+			}
+			return err
+		}
+		allResources[wl.GetID()] = wl
+		// k8sResourcesMap[<ID>] = workload
+		(*k8sResourcesMap)[fmt.Sprintf("%s/%s/%ss", wl.GetApiVersion(), wl.GetNamespace(), strings.ToLower(wl.GetKind()))] = []string{wl.GetID()}
```
Comment (dwertent) [inline resourcehandler/k8sresources.go:95]: You should use built in functions for converting single kind to plural.
consider calling [this function](https://github.com/armosec/k8s-interface/blob/6f24b7195a92aa1fb71927bc52c5d124d2afa757/k8sinterface/k8sdiscovery.go#L131)

File: resultshandling/printer/prettyprinter.go [hunk 3]
```diff
@@ -158,6 +158,9 @@ func (printer *PrettyPrinter) printGroupedResources(workloads map[string][]Workl
 	if rsc, ok := workloads["Group"]; ok {
 		printer.printGroupedResource(indent, "Group", rsc)
 	}
+	if rsc, ok := workloads["ClusterDescription"]; ok {
```
Comment (dwertent) [inline resultshandling/printer/prettyprinter.go:161]: The `ClusterDescription` is the value not the key

File: resultshandling/printer/summeryhelpers.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ func groupByNamespaceOrKind(resources []WorkloadSummary, status func(workloadSum
 }
 
 func isKindToBeGrouped(kind string) bool {
-	if kind == "Group" || kind == "User" {
+	if kind == "Group" || kind == "User" || kind == "CloudProvider" || kind == "ClusterDescription" {
```
Comment (dwertent) [inline resultshandling/printer/summeryhelpers.go:32]: `CloudProvider` Is a kind we do not support.
Please refer to the [const](https://github.com/armosec/k8s-interface/blob/master/cloudsupport/cloudproviderconfiguration.go#L27) and not `"ClusterDescription"`

File: resourcehandler/k8sresources.go (hunk 2 above)
Comment (dwertent) [reply resourcehandler/k8sresources.go:95]: Also, you do not want to use the namespace, only group, version and kind. This are the resources mentioned in the match object

File: resultshandling/printer/summeryhelpers.go (hunk 4 above)
Comment (Daniel-GrunbergerCA) [reply resultshandling/printer/summeryhelpers.go:32]: Otherwise it will be printed as 'namespace'

File: resourcehandler/k8sresources.go (hunk 2 above)
Comment (Daniel-GrunbergerCA) [reply resourcehandler/k8sresources.go:95]: I see it returns a list... you want me to use it anyway?

File: resourcehandler/k8sresources.go (hunk 1 above)
Comment (Daniel-GrunbergerCA) [reply resourcehandler/k8sresources.go:83]: It is, here we want to display custom error for developer and I thoght it's not supposed to be on the pkg. I can put it there tho

**Pull request closed without merging**